:seedling: Utilize tilt-provider.yaml and Tilt deploy approach from the Cluster API (#726)
kubernetes-sigs/cluster-api-operator: docs/local-development.md

# Local Development
Tilt is favoured by most Cluster API projects for local development, it offers a simple way of creating a local development environment.
Cluster API includes its own Tiltfile that can be used to run Cluster API Operator on a local Kind cluster.

## Clone the Cluster API repository

> Clone the Cluster API repository on the same level folder where Cluster API Operator located:

Reviewer: ```suggestion
Clone the Cluster API repository in the same directory as the Cluster API Operator:
```

```bash

> $ git clone https://github.com/kubernetes-sigs/cluster-api.git

Reviewer: you should not need `$` since it is in bash quotes already:
```suggestion
git clone https://github.com/kubernetes-sigs/cluster-api.git
```

```

> Afterward your folder structure should look like this:

Reviewer: ```suggestion
Afterward, your folder structure should look like as follows:
```

```
some-folder/
├── cluster-api
└── cluster-api-operator
```

## Set up Tilt settings in `cluster-api` folder

Refer to [this guide](https://cluster-api.sigs.k8s.io/developer/core/tilt.html) to set up Tilt for Cluster API.

> In particular, for our purposes we only need to set up `tilt-settings.yaml` in Cluster API to enable Cluster API Operator. Add the following fields to the lists in `tilt-settings.yaml`:

Reviewer: ```suggestion
For our use case, you only need to configure `tilt-settings.yaml` in the `cluster-api` directory to enable the Cluster API Operator. Add the following fields to the corresponding lists in `tilt-settings.yaml`:
```

```yaml
provider_repos:
- "../cluster-api-operator"
enable_providers:
- capi-operator
enable_core_provider: false
```

## Run Tilt

From `cluster-api` folder run:

```bash

> $ make tilt-up

Reviewer: I am not sure about this, do you really need `make` here?
```suggestion
make tilt-up
```

> make tilt-up

Reviewer 2: ```suggestion
make docker-build-e2e # Use locally built CAPI images
make tilt-up
```

Author: Good point, added.

```

> That's it! Tilt will automatically reload the deployment to your local cluster every time you make a code change and able to debug deployed code.

Reviewer: ```suggestion
That's it! Tilt will automatically reload the deployment in your local cluster whenever you make code changes, allowing you to debug the deployed code in real time.
```
